packages/astro/src/runtime/server/escape.ts

```diff
@@ -1,6 +1,6 @@
 const entities = { '"': 'quot', '&': 'amp', "'": 'apos', '<': 'lt', '>': 'gt' } as const;
 
-export const escapeHTML = (string: any) => string.replace(/["'&<>]/g, (char: keyof typeof entities) => '&' + entities[char] + ';');
+export const escapeHTML = (str: any) => str != null ? String(str).replace(/["'&<>]/g, (char: string) => '&' + entities[char as keyof typeof entities] + ';') : str;
```

Comment:
these functions are getting pretty complex for a 1-liner. Mind converting this and `unescapeHTML` into larger functions?

```
export function escapeHTML(val: any) {
  // COMMENT EXPLAINING WHY
  if (val == null) {
    return val;
  }
  // otherwise, cast to a string and escape HTML entities
}
```

Also, is this now doing more than just escaping HTML? This may be better named as "sanitizeHTML" or "serializeHTML" or "stringifyHTML". Ditto for `unescapeHTML`

Also (out of scope for this PR, but) I googled around and found people saying basically "don't write your own html sanitization". This package readme goes into detail a bit: https://www.npmjs.com/package/html-escaper. I think there's a case to replace our custom escaping with a more battle-tested package and reduce our overall maintenance.

Reply:
Actually, Chris and I are debugging some escaping issues in the new docs site, it looks like expressions are getting double escaped. Okay to leave these as-is in this PR then, and I'll base my fix to that issue off of your branch here.